kubetest/kind/kind.go

```diff
@@ -121,33 +147,80 @@ func (k *Kind) setKubeConfigEnv() error {
 	return nil
 }
 
-// Up builds kind and the node image and then deploys a cluster based on the kind config.
-func (k *Kind) Up() error {
-	// TODO(neolit123): fetch a stable release binary and stash it in a kubetest specific dir.
-	// This should be the default option instead of building kind from source.
+// prepareKindBinary either builds kind from source or pulls a binary from GitHub.
+func (k *Kind) prepareKindBinary() error {
+	// TODO(neolit123): before downloading check if a binary already exists;
+	//   we can either match its version against the release tag or match SHAs post download.
+	// TODO(neolit123): support other architectures.
+	// TODO(neolit123): support other OS.
```

Comment:
i can send follow up PR once we decide on how to approach this.
should using `runtime.GOOS`, `runtime.GOARCH` suffice here?

this makes sense for local use, but AFAIK in test jobs this will end up always being `linux-amd64`.

Reply:
yeah, let's use those so it works locally too.